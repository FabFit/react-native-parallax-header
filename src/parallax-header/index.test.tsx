import React from 'react';
import { render } from '@testing-library/react-native';
import ParallaxHeader from './index';

test('[default] ParallaxHeader matches the snapshot', () => {
  const { baseElement, queryByTestId } = render(<ParallaxHeader />);

  expect(queryByTestId('hero-image')).toBeNull();
  expect(baseElement).toMatchSnapshot();
});

test('[custom height] ParallaxHeader matches the snapshot', () => {
  const { baseElement, queryByTestId } = render(
    <ParallaxHeader maxHeight={700} minHeight={300} />
  );

  expect(queryByTestId('hero-image')).toBeNull();
  expect(baseElement).toMatchSnapshot();
});

test('[heroImage] ParallaxHeader matches the snapshot', () => {
  const { baseElement, getByTestId } = render(
    <ParallaxHeader
      heroImage={{ uri: 'https://source.unsplash.com/300x500/?pilates,yoga' }}
    />
  );

  expect(getByTestId('hero-image')).toBeTruthy();
  expect(baseElement).toMatchSnapshot();
});
