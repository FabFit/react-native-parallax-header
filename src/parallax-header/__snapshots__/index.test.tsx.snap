// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[custom height] ParallaxHeader matches the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <ScrollView
      scrollEventThrottle={1}
      showsVerticalScrollIndicator={false}
    >
      <View
        style={
          Object {
            "marginTop": 700,
          }
        }
      />
    </ScrollView>
    <View
      style={
        Object {
          "backgroundColor": "rgba(45, 51, 59, 1)",
          "height": 700,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    />
  </View>
</View>
`;

exports[`[default] ParallaxHeader matches the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <ScrollView
      scrollEventThrottle={1}
      showsVerticalScrollIndicator={false}
    >
      <View
        style={
          Object {
            "marginTop": 550,
          }
        }
      />
    </ScrollView>
    <View
      style={
        Object {
          "backgroundColor": "rgba(45, 51, 59, 1)",
          "height": 550,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    />
  </View>
</View>
`;

exports[`[heroImage] ParallaxHeader matches the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <ScrollView
      scrollEventThrottle={1}
      showsVerticalScrollIndicator={false}
    >
      <View
        style={
          Object {
            "marginTop": 550,
          }
        }
      />
    </ScrollView>
    <View
      style={
        Object {
          "backgroundColor": "rgba(45, 51, 59, 1)",
          "height": 550,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://source.unsplash.com/300x500/?pilates,yoga",
          }
        }
        style={
          Object {
            "height": 550,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "resizeMode": "cover",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        testID="hero-image"
      />
    </View>
  </View>
</View>
`;
